modelforge/dataset/utils.py

```diff
@@ -1 +1,112 @@
-# generates and retrieves hdf5 files from zenodo, defines the interaction with the hdf5 file 
\ No newline at end of file
+# generates and retrieves hdf5 files from zenodo, defines the interaction with the hdf5 file
+from torch.utils.data import random_split
+import torch
+from typing import List
+import numpy as np
+from loguru import logger
+
+
+class SplittingStrategy:
+    def __init__(
+        self,
+        seed: int,
+    ):
+        self.seed = seed
+        torch.manual_seed(self.seed)
+
+    def split():
+        raise NotImplementedError
+
+
+class RandomSplittingStrategy(SplittingStrategy):
+    def __init__(self, seed: int = 42, split: List[float] = [0.8, 0.1, 0.1]):
+        super().__init__(seed)
+        self.train_size = split[0]
+        self.val_size = split[1]
+        self.test_size = split[2]
+        assert np.isclose(sum(split), 1.0)
+
+    def split(self, dataset) -> List[List[int]]:
+        torch.Generator().manual_seed(self.seed)
+
+        logger.debug(f"Using random splitting strategy with seed {self.seed} ...")
+        logger.debug(
+            f"Splitting dataset into {self.train_size}, {self.val_size}, {self.test_size} ..."
+        )
+
+        train_d, val_d, test_d = random_split(
+            dataset,
+            lengths=[self.train_size, self.val_size, self.test_size],
+        )
```

Comment:
```suggestion
             generator= torch.Generator().manual_seed(self.seed))
```

Reply:
I think the reason the test is failing is that you aren't passing the generator to random_split.   So setting the seed in line 30  doesn't end up doing anything.

Reply:
ah, thanks for catching this